infra/scripts/bash-helpers.sh

```diff
@@ -13,32 +13,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.!/bin/bash
 
+[ -n "$DEBUG" ] && set -x
+
+BASE_DIR=$(dirname $(readlink -f "$BASH_SOURCE"))
+REPO_DIR=$(pwd "${BASE_DIR}/../../")
 OS=$(uname | tr '[:upper:]' '[:lower:]')
 
-tls () {
-    unset TLS_OPTS
-}
+# global settings that reduce friction
+if [ -z "${GOVC_TLS_KNOWN_HOSTS}" ]; then
+    export GOVC_TLS_KNOWN_HOSTS=~/.govmomi/known_hosts
+fi
 
-no-tls () {
-    export TLS_OPTS="--no-tls"
+init-profile () {
+    unset-vic
+    vch_name=${FUNCNAME[1]}
 }
 
 unset-vic () {
-    unset MAPPED_NETWORKS NETWORKS IMAGE_STORE DATASTORE COMPUTE VOLUME_STORES IPADDR GOVC_INSECURE TLS THUMBPRINT OPS_CREDS VIC_NAME
+    unset TARGET_URL MAPPED_NETWORKS NETWORKS IMAGE_STORE DATASTORE COMPUTE VOLUME_STORES IPADDR GOVC_INSECURE TLS THUMBPRINT OPS_CREDS VIC_NAME PRESERVE_VOLUMES
+
+    unset vsphere datacenter user password opsuser opspass opsgrant timeout compute datastore dns publicNet publicIP publicGW bridgeNet bridgeRange
+    unset clientNet clientIP clientGW managementNet managementIP managementGW tls volumestores preserveVolumestores containernet
 }
 
 vic-path () {
     echo "${GOPATH}/src/github.com/vmware/vic"
 }
 
 vic-create () {
+    vicProfileTranscode
+
     base=$(pwd)
     (
         cd "$(vic-path)"/bin || return
-        "$(vic-path)"/bin/vic-machine-"$OS" create --target="$GOVC_URL" "${OPS_CREDS[@]}" --image-store="$IMAGE_STORE" --compute-resource="$COMPUTE" "${TLS[@]}" ${TLS_OPTS} --name="${VIC_NAME:-${USER}test}" "${MAPPED_NETWORKS[@]}" "${VOLUME_STORES[@]}" "${NETWORKS[@]}" ${IPADDR} ${TIMEOUT} --thumbprint="$THUMBPRINT" "$@"
+        "$(vic-path)"/bin/vic-machine-"$OS" create --target="$TARGET_URL" "${OPS_CREDS[@]}" --image-store="$IMAGE_STORE" --compute-resource="$COMPUTE" "${TLS[@]}" ${TLS_OPTS} --name="${VIC_NAME:-${USER}test}" "${MAPPED_NETWORKS[@]}" "${VOLUME_STORES[@]}" "${NETWORKS[@]}" ${IPADDR} ${TIMEOUT} --thumbprint="$THUMBPRINT" "$@"
```

Comment:
Would it make sense to set a debug level by default?

Reply:
I don't - it's very simple to add refinement onto the end of the function, e.g:
```
vic-create --debug=2`
```